postgres/server/init.sls

```diff
@@ -22,11 +16,16 @@ include:
 postgresql-server:
   pkg.installed:
     - pkgs: {{ pkgs }}
-{%- if postgres.use_upstream_repo %}
+  {%- if postgres.pkg_repo.name %}
+    - fromrepo: {{ postgres.pkg_repo.name }}
+  {%- endif %}
+  {%- if postgres.use_upstream_repo %}
     - refresh: True
     - require:
       - pkgrepo: postgresql-repo
-{%- endif %}
+
+    {%- do includes.append('postgres.upstream') %}
```

Comment:
We have already substituted the `includes` list above, so this line has no effect and the `postgres.upstream` states would never be included. It is always better to keep definition and logic around `includes` statements at the top of SLS file. That looks simpler and makes debugging easier. By removing a single `if ` statement you don't make the code more readable or "optimized".

Reply:
In support of opinions in #125, its good idea to removing some Jinja in SLS/YML files. These sentiments, which I tend to agree with, can only be addressed in baby steps. No issue here.

I moved this includes down few lines in the code - its the first occurance so surely takes effect.

Reply:
I saw you did it differently in the `client.sls`. In THAT case it would work, despite not looking pretty though...

Please double check: moving `do includes.append...` statement here would break calling the single state like `salt-call state.apply postgres.server`. The `postgres.upstream` SLS would be never included this way.

Reply:
The server SLS is more tricky due to `includes` Jinja.  And reviewing this, maybe the following Jinja must move below  `postgresql.server.pkg.installed` state so indent(2) is applied?
```
{%- if includes -%}
include:
  {{ includes|yaml(false)|indent(2) }}
{%- endif %}
```
The usage of `salt.apply` is new for me. I'll will report back.